examples/github-stars-example/README.md

```diff
@@ -0,0 +1,307 @@
+# Overview
+
+In this example we will use Versatile Data Kit to create Streamlit web app that displays GitHub Star History:
+
+[<img src="https://user-images.githubusercontent.com/11227374/192750161-dfca62f0-5ea3-4300-8887-4413c8dbbf05.png" width="500"/>](Github)
+
+The job will insert the data from GitHub API into local sqlite table and create a Streamlit app. It also edits the default streamlit color schema.
+
+Before you continue, make sure you are familiar with the
+[Getting Started](https://github.com/vmware/versatile-data-kit/wiki/Getting-Started) section of the wiki.
+
+How to run this job locally
+-----
+To run this job, follow the steps:
+
+1. Clone the repo
+```
+git clone https://github.com/vmware/versatile-data-kit.git
+```
+2. Install vdk
+```
+pip install -U pip setuptools wheel
+pip install quickstart-vdk
+```
+3. Now do vdk create, enter the job name, team and location
+```
+vdk create
+
+Job Name: local-github-stars
+Job Team: test
+Path to where sample data job will be created locally [/home/gita/vdk]:
+Data Job with name local-github-stars created locally in /home/gita/vdk/local-github-stars.
+```
+4. Clear the new directory and copy example files to it
+```
+rm local-github-stars/*
+cp versatile-data-kit/examples/github-stars-example/github-stars-example-job/* local-github-stars/
+```
+5. Add your GitHub token and repo name to the file 00_properties.py
+```
+vim local-github-stars/00_properties.py
```

Comment:
IMO it is strange that you would be expected to add a secret in a file. 
As opposed to it reading it from an env_variable or something similar to that.

Reply:
Is it an issue? Both ways are ok. I would prefer having it in the file.

Reply:
I think it is an issue. 
Even though this is a toy example it is supposed to guide others in how to write production jobs and you would never be expected to put a secret in a file in a production job. 

I think you should probably load it from the filesystem using @ivakoleva's plugin?

Reply:
Storing credentials as part of the source code is a [bad](https://pvs-studio.com/en/docs/warnings/v5601/) [practice](https://medium.com/twodigits/keep-passwords-out-of-source-code-why-and-how-e84f9004815a) so a) we definitely should not encourage it b) it will make a bad impression for our project if someone who knows about this stuff is looking through those. 

The way our users store credentials is to use Properties API but they do it using CLI or API/UI : 

So in the 00_properties.py file, we should only set the API URL perhaps and instruct users to use 
```
export properties_default_type=fs-properties-client
vdk properties --set-secret "token" # will be prompted for the actual token
```
See vdk properties --help for more info.

Reply:
For me it makes sense the way it is implemented now. However a fresh view is probably more valuable in terms of presenting examples to new users.

maybe (1) mark vdk-properties-fs for local dev/demo purposes only (requirements.txt, config.ini); 
and (2) instead of having a file with placeholder -> add instructions how to initialize the properties before data job run/deploy - using the VDK CLI commands, so that the data job reuses those secrets after? that includes toggling the properties client config (between the new and default plugin)

Reply:
Another option is to use arguments : 
```
vdk run --arguments '{"token":"xxx", "repo_path":"versatiledatakit/foo"}'
``` 

Arguments are then accessed in the job step using `job_input.get_arguments()` method 


Properties is better because `vdk properties --set-secret `  prompts you for the secret. While in arguments you set it in the command line.  Supplying passwords via command line argument is considered bad security practices because the password itself is leaked into the process table and/or in the shell history file. If curious you can read more details here - https://www.netmeister.org/blog/passing-passwords.html 

Another reason it's better is that you set it once only no need to supply them every time. 
Now of course local fs properties do store them in a file. But if one uses Control Service it won't. Besides in the future we may encrypt the file if FS properties are really used more frequently. 

In any case, the passing arguments approach is still better than setting them in the source file. 

This is more of FYI sharing what all the options are. I think we should use the properties CLI for this example.

Reply:
When I use properties CLI, the properties get set, but the job fails with an error: None
I don't know how to fix it, maybe someone else can test? I need some help with this. 

```
    {
      "name": "30_stargazers.py",
      "type": "python",
      "start_time": "2022-11-09T10:44:25.686497",
      "end_time": "2022-11-09T10:44:25.715008",
      "status": "error",
      "details": "An exception occurred, exception message was: An error in data job code  occurred. The error should be resolved by User Error. Here are the details:\n  WHAT HAPPENED : Data Job step 30_stargazers.py completed with error.\nWHY IT HAPPENED : An exception occurred, exception message was: None\n   CONSEQUENCES : I will not process the remaining steps (if any), and this Data Job execution will likely be marked as failed.\nCOUNTERMEASURES : See exception and fix the root cause, so that the exception does not appear anymore.",
      "exception": {
        "is_logged": true,
        "cause_exception_name": "AssertionError",
        "cause_exception": {
          "is_logged": true
        }
      },
      "blamee": "User Error",
      "exception_name": "UserCodeError"
    }
  ],
  "exception": null,
  "blamee": "User Error",
  "exception_name": "UserCodeError"
}

Reply:
In config,ini there is 

`properties_default_type=fs-properties-client` 

This controls what is the source for the properties of your data job. 

When you are setting them manually with `vdk properties --set`, if the same is not set as env variable you may be setting them somewhere else (by default control service perhaps)> 

Would something like this work ?
```
export properties_default_type=fs-properties-client
vdk properties --set ...
```

Reply:
I fixed the code, maybe someone else can test it. I can't run this job, it still fails with error; None even after setting the properties_default_type to fs-properties-client with export. 
I don't have the control service installed where I'm running the job.

Reply:
Is this resolved?

Reply:
No, it doesn't work for me

Reply:
Looking at this now